tests/e2e/default/contacts/muting-contact-using-form.wdio-spec.js

```diff
@@ -0,0 +1,75 @@
+const utils = require('../../../utils');
+const commonPage = require('../../../page-objects/default/common/common.wdio.page');
+const loginPage = require('../../../page-objects/default/login/login.wdio.page');
+const userFactory = require('../../../factories/cht/users/users');
+const placeFactory = require('../../../factories/cht/contacts/place');
+const personFactory = require('../../../factories/cht/contacts/person');
+const modalPage = require('../../../page-objects/default/common/modal.wdio.page');
+const contactPage = require('../../../page-objects/default/contacts/contacts.wdio.page');
+const genericForm = require('../../../page-objects/default/enketo/generic-form.wdio.page');
+const deathReportForm = require('../../../page-objects/default/enketo/death-report.page');
+const undoDeathReportForm = require('../../../page-objects/default/enketo/undo-death-report.page');
+
+describe('Mute/Unmute contacts using a specific form.', () => {
+  const places = placeFactory.generateHierarchy();
+  const healthCenter = places.get('health_center');
+  const offlineUser = userFactory.build({ place: healthCenter._id, roles: ['chw'] });
+  const person = personFactory.build({ parent: {_id: healthCenter._id, parent: healthCenter.parent} });
+  const mutePerson = personFactory.build({ 
+    patient_id: 12345, 
+    name: 'mutedPerson', 
+    muted: new Date(),
+    parent: {_id: healthCenter._id, parent: healthCenter.parent} 
+  });
+
+  const settings_mute = {
+    transitions: {
+      death_reporting: true,
+      muting: true
+    },
+    muting: {
+      mute_forms: ['death_report'],
+      unmute_forms: ['undo_death_report']
+    }
+  };
+
+  before(async () => {
+    await utils.saveDocs([...places.values(), person, mutePerson]);
+    await utils.updateSettings(settings_mute, true);
+    await utils.createUsers([offlineUser]);
+    await loginPage.login(offlineUser);
+  });
+
+  it('Should mute a contact using the defined mute_forms (death_report).', async () => {
+    await commonPage.goToPeople(person._id);
+    await contactPage.createNewAction('Death report');
+    await deathReportForm.submitDeathReport();
+    await commonPage.sync(true);
+
+    expect(await (await contactPage.contactMuted()).isDisplayed()).to.be.true;
+
+  });
+
+  it('Should unmute a contact using the defined unmute_forms (undo_death_report).', async () => {
+    await commonPage.goToPeople(person._id);
+    await contactPage.createNewAction('Undo death report');
+    await undoDeathReportForm.setConfirmUndoDeathOption();
+    await genericForm.submitForm();
+    await commonPage.waitForPageLoaded();
+    await commonPage.sync(true);
+
+    expect(await (await contactPage.contactMuted()).isDisplayed()).to.be.false;
+
+  });
+
+  it('Should show a popup when trying to submit a non-unmuting form against a muted contact', async () => {
+    await utils.revertSettings(true);
```

Comment:
Why do we need to revert the settings? this mean the `setting_mute` setting is not going to be active for this scenario?

Reply:
Yes, because what we are trying to check here is that if we have a muted person, and we want to submit a form,  a form that is not configured in the muting settings, a pop-up should be displayed saying that you are trying to submit a form against a muted person.

Reply:
I want to make sure I understand. We will have the same results keeping the `setting_mute` setting active, and using a form, for example, pregnancy, unrelated to the `setting_mute`?

Reply:
That is correct.
The forms that are "affected" by the `muting` setting are the ones assigned to `mute_forms` or `unmute_forms`.
I can show you how to test that manually, maybe that can help. It was also hard for me to understand it at the beginning, I had to do a lot of different tests. 
[Here](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/transitions/#muting) is more information

Reply:
Thanks @tatilepizs I think it is clearer now. It is a really interesting feature to try edge cases in the future. Great job!

Reply:
Thank you! 😊